### PR TITLE
refactor: simplify code by removing some recursivity

### PR DESCRIPTION
**Pull request by Jisay**

### CODE REVIEW

File: src/helpers/ovh-contacts.helpers.js [hunk 1]
```diff
@@ -146,126 +140,127 @@ export default class OvhContactsHelper {
...
+          // set default value from nicRule to contactRule
+          set(contactRule, 'defaultValue', nicRule.defaultValue);
+          // set initial value from initialValues param
+          set(contactRule, 'initialValue', get(initialValues, contactRule.name));
+          // set regular expression from nicRule to contactRule
+          set(contactRule, 'regularExpression', nicRule.regularExpression);
+          // set prefix from nicRule to contactRule
+          set(contactRule, 'prefix', nicRule.prefix);
+          // set example from nicRule to contactRule
+          set(contactRule, 'examples', nicRule.examples);
```
Comment (marie-j) [inline src/helpers/ovh-contacts.helpers.js]:
```suggestion
          Object.assign(contactRule,{
            defaultValue: nicRule.defaultValue,
            initialValue: get(initialValues, contactRule.name), 
            ...
          });
```

File: src/ovh-contacts.service.js [hunk 2]
```diff
@@ -155,49 +149,42 @@ export default class OvhContactsService {
       })
       .then(rulesOptions => this.$q.all({
         apiSchemas: this.getMeSchemas(),
-        creationRules: this.OvhApiNewAccount.v6().rules({}, rulesOptions).$promise,
+        nicCreationRules: this.OvhApiNewAccount.v6().rules({}, rulesOptions).$promise,
```
Comment (marie-j) [inline src/ovh-contacts.service.js]: It would be better to have an action verb like `getRules` than just `rules` imo

Comment (Jisay) [reply src/ovh-contacts.service.js]: In fact... It's a post to the API :-)

File: src/helpers/ovh-contacts.helpers.js (hunk 1 above)
Comment (Jisay) [reply src/helpers/ovh-contacts.helpers.js]: Good catch!

File: src/form/ovh-contact-form.controller.js [hunk 3]
```diff
@@ -164,11 +154,22 @@ export default class OvhContactsFormCtrl {
...
         if (!this.initialDefaultValues) {
           this.initialDefaultValues = {};
-          OvhContactsFormCtrl.traverseRules(rules, (rule, ruleKey) => {
-            if (ruleKey !== 'phoneCountry' && (rule.defaultValue || rule.initialValue)) {
-              set(this.initialDefaultValues, ruleKey, rule.defaultValue);
-              set(this.model, ruleKey, rule.initialValue || rule.defaultValue);
-              if (ruleKey === 'address.country') {
+
+          forIn(rules, (ruleValue) => {
+            const isPhoneCountry = ruleValue.name !== 'phoneCountry';
```
[the remark this answers is not in the document]
Comment (Jisay) [reply src/form/ovh-contact-form.controller.js]: Ooooops :-)

**Comment by jleveugle** [comment]
Hello @Jisay,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Pull request closed without merging**